EventFilter/L1TRawToDigi/src/OmtfLinkMappingRpc.cc

```diff
@@ -0,0 +1,127 @@
+#include "EventFilter/L1TRawToDigi/interface/OmtfLinkMappingRpc.h"
+
+#include <fstream>
+
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "CondFormats/RPCObjects/interface/RPCReadOutMapping.h"
+#include "CondFormats/RPCObjects/interface/RPCEMap.h"
+#include "CondFormats/DataRecord/interface/RPCEMapRcd.h"
+#include "CondFormats/RPCObjects/interface/DccSpec.h"
+#include "CondFormats/RPCObjects/interface/LinkBoardElectronicIndex.h"
+#include "CondFormats/RPCObjects/interface/LinkBoardPackedStrip.h"
+#include "CondFormats/RPCObjects/interface/TriggerBoardSpec.h"
+#include "CondFormats/RPCObjects/interface/LinkBoardSpec.h"
+
+#include "CondFormats/DataRecord/interface/RPCOMTFLinkMapRcd.h"
+#include "CondFormats/RPCObjects/interface/RPCAMCLinkMap.h"
+
+namespace omtf {
+
+bool lessLinkBoardElectronicIndex::operator() (const LinkBoardElectronicIndex & o1, const LinkBoardElectronicIndex & o2) {
+  if (o1.dccId < o2.dccId) return true;
+  else if (o1.dccId == o2.dccId && o1.dccInputChannelNum < o2.dccInputChannelNum) return true;
+  else if (o1.dccId == o2.dccId && o1.dccInputChannelNum == o2.dccInputChannelNum && o1.tbLinkInputNum < o2.tbLinkInputNum) return true;
+  else return false;
+}
+
+
+MapEleIndex2LBIndex translateOmtf2Pact(const RpcLinkMap & omtfLink2Ele, const RPCReadOutMapping* pactCabling) {
+  MapEleIndex2LBIndex omtf2rpc;
+
+  std::vector<const DccSpec*> dccs = pactCabling->dccList();
+  for (auto it1 : dccs) {
+    const std::vector<TriggerBoardSpec> & rmbs = it1->triggerBoards();
+    for (auto const & it2 : rmbs) {
+      const  std::vector<LinkConnSpec> & links = it2.linkConns();
+      for (auto const & it3 : links) {
+        const  std::vector<LinkBoardSpec> & lbs = it3.linkBoards();
+        for (std::vector<LinkBoardSpec>::const_iterator it4=lbs.begin(); it4 != lbs.end(); ++it4) {
+
+          try {
+            std::string lbNameCH = it4->linkBoardName();
+            std::string lbName = lbNameCH.substr(0,lbNameCH.size()-4);
+            const std::vector<EleIndex> & omtfEles = omtfLink2Ele.omtfEleIndex(lbName);
+//          std::cout <<"  isOK ! " <<  it4->linkBoardName() <<" has: " << omtfEles.size() << " first: "<< omtfEles[0] << std::endl;
+            LinkBoardElectronicIndex rpcEle = { it1->id(), it2.dccInputChannelNum(), it3.triggerBoardInputNumber(), it4->linkBoardNumInLink()};
+            for ( const auto & omtfEle : omtfEles ) omtf2rpc[omtfEle]= rpcEle;
+          }
+          catch(...) { ; } // std::cout << "exception! "<<it4->linkBoardName()<< std::endl; }
```

Comment:
Why was this egregious, unnecessary use of try/catch allowed to persist? It is effectively impossible to debug any real exception now.